### PR TITLE
[FLOC-1772] Split deletes.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-1772

### CODE REVIEW

File: flocker/node/agents/blockdevice.py [hunk 1]
```diff
@@ -1832,7 +1805,7 @@ def _calculate_deletes(self, local_node_state, configured_manifestations,
             volume = _blockdevice_volume_from_datasetid(volumes, dataset_id)
             if (volume is not None and
```
Comment (itamarst) [inline flocker/node/agents/blockdevice.py]: This seems like it needs fixing: we don't want to destroy things that are in local manifestations, we want to destroy things in UNATTACHED state.

File: flocker/node/agents/test/test_blockdevice.py [hunk 2]
```diff
@@ -1542,10 +1577,10 @@ def test_no_delete_if_leased(self):
 
     def test_deleted_dataset_volume_unmounted(self):
```
Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: This seems like a duplicate test now?

Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: Ah, no. Issue is the docstring.

File: flocker/node/agents/test/test_blockdevice.py [hunk 3]
```diff
@@ -1542,10 +1577,10 @@ def test_no_delete_if_leased(self):
...
         """
-        ``DestroyBlockDeviceDataset`` is a compound state change that first
-        attempts to unmount the block device.
-        Therefore do not calculate deletion for blockdevices that are not
-        manifest.
+        If the configuration indicates a dataset with a primary manifestation
+        on the node has been deleted and the volume associated with that
+        dataset is mounted, ``BlockDeviceDeployer.calculate_changes`` returns a
```
Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: "unattached", not "mounted"

File: flocker/node/agents/test/test_blockdevice.py [hunk 4]
```diff
@@ -1902,6 +1931,31 @@ def test_unmount_manifestation(self):
             ])
         )
 
+    def test_unmount_deleted_manifestation(self):
```
Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: Is this another duplicate?

File: flocker/node/agents/test/test_blockdevice.py [hunk 5]
```diff
@@ -2428,6 +2482,47 @@ def test_detach_manifestation(self):
...
+        # some attached volumes.
+        node_state = NodeState(
+            uuid=self.NODE_UUID, hostname=self.NODE,
+            applications={},
+            manifestations={},
+            devices={self.DATASET_ID: FilePath(b"/dev/xda")},
+            paths={},
+        )
+
+        # Give it a configuration that says no datasets should be manifest on
```
Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: This comment doesn't match the code.

File: flocker/node/agents/blockdevice.py (hunk 1 above)
Comment (itamarst) [inline flocker/node/agents/blockdevice.py]: Getting a bunch of test failures probably caused by this.

**Comment by itamarst** [comment]
Thanks for fixing this! Please address the above and resubmit.

File: flocker/node/agents/test/test_blockdevice.py (hunk 2 above)
Comment (tomprince) [inline flocker/node/agents/test/test_blockdevice.py]: Maybe? I think perhaps only with the test you commented on below. I'm not sure if I'm smart enough to disentangle the extraneous details to see which tests are necessary. These tests will all be superfluous once the code is refactored, so I'm inclined to leave them in for the moment.

**Pull request merged**